spec/ics-027-interchain-account/README.md

```diff
@@ -0,0 +1,283 @@
+
+- The other module is bound to the "interchain account" port.
+- The channel being created is ordered.
+- The version string is empty.
+```typescript
+function onChanOpenInit(
+  order: ChannelOrder,
+  connectionHops: [Identifier],
+  portIdentifier: Identifier,
+  channelIdentifier: Identifier,
+  counterpartyPortIdentifier: Identifier,
+  counterpartyChannelIdentifier: Identifier,
+  version: string) {
+  // only ordered channels allowed
+  abortTransactionUnless(order === ORDERED)
+  // only allow channels to "interchain-account" port on counterparty chain
+  abortTransactionUnless(counterpartyPortIdentifier === "interchain-account")
+  // version not used at present
+  abortTransactionUnless(version === "")
+}
+```
+```typescript
+function onChanOpenTry(
+  order: ChannelOrder,
+  connectionHops: [Identifier],
+  portIdentifier: Identifier,
+  channelIdentifier: Identifier,
+  counterpartyPortIdentifier: Identifier,
+  counterpartyChannelIdentifier: Identifier,
+  version: string,
+  counterpartyVersion: string) {
+  // only ordered channels allowed
+  abortTransactionUnless(order === ORDERED)
+  // version not used at present
+  abortTransactionUnless(version === "")
+  abortTransactionUnless(counterpartyVersion === "")
+  // only allow channels to "interchain-account" port on counterparty chain
+  abortTransactionUnless(counterpartyPortIdentifier === "interchain-account")
+}
+```
+```typescript
+function onChanOpenAck(
+  portIdentifier: Identifier,
+  channelIdentifier: Identifier,
+  version: string) {
+  // version not used at present
+  abortTransactionUnless(version === "")
+  // port has already been validated
+}
+```
+```typescript
+function onChanOpenConfirm(
+  portIdentifier: Identifier,
+  channelIdentifier: Identifier) {
+  // accept channel confirmations, port has already been validated
+}
+```
+```typescript
+function onChanCloseInit(
+  portIdentifier: Identifier,
+  channelIdentifier: Identifier) {
+  // no action necessary
+}
+```
+```typescript
+function onChanCloseConfirm(
+  portIdentifier: Identifier,
+  channelIdentifier: Identifier) {
+  // no action necessary
+}
+```
+### **Packet relay**
+
+In plain English, between chains `A` and `B`. It will describe only the case that chain A wants to register an Interchain account on chain B and control it. Moreover, this system can also be applied the other way around. In these set of examples, we follow Cosmos-SDK's modularity rule and its `msg` system.
+```typescript
+interface ChainAccountTx {
+  msgs: Msg[]
+}
+```
+```typescript    
+function onSendPacket(packet: Packet) {  
+  // Sending chain can do preliminary work (i.e. create a log of transactions sent) if needed.
+}
+```
+```typescript
+function onRecvPacket(packet: Packet): bytes {
+  if (packet.data is RunTxPacketData) {
+    ChainAccountTx data = {}
+    // Decode transaction request
+    codec.unmarshal(packet.data.txBytes, &data)
```

Comment:
You don't need to bother with this in pseudocode

Reply:
Ok